tests/message_mattermost_manager_test.php

```diff
@@ -0,0 +1,167 @@
+
+    /**
+     * Function to test redirect_uri
+     */
+    public function test_redirect_uri() {
+        $mattermostmanager = new message_mattermost\manager();
+        $outputuri = $mattermostmanager->redirect_uri();
+
+        // Verify redirect_uri.
+        $expecteduri = 'https://www.example.com/moodle/'
+        .'message/output/mattermost/mattermostconnect.php';
+        $this->assertEquals($expecteduri, $outputuri);
+    }
+
+    /**
+     * Function to test config_form when notificationstate is enabled
+     */
+    public function test_config_form_with_notificationstate_enabled() {
+        $config = $this->set_plugin_config();
+        $mattermostmanager = new message_mattermost\manager($config);
+
+        // Create a new test user.
+        $user = $this->create_test_user_and_reset_state();
+
+        $outputconfigbutton = $mattermostmanager->config_form('', $user->id);
+        $expectedconfigbutton = 'Disable Mattermost Notifications';
+
+        $this->assertTrue(strpos($outputconfigbutton, $expectedconfigbutton) !== false);
+    }
+
+    /**
+     * Function to test config_form when notificationstate is disabled
+     */
+    public function test_config_form_with_notificationstate_disabled() {
+        $config = $this->set_plugin_config(0);
+        $mattermostmanager = new message_mattermost\manager($config);
+
+        // Create a new test user.
+        $user = $this->create_test_user_and_reset_state();
+
+        // Test configbutton when plugin is disabled.
+        $outputconfigbutton = $mattermostmanager->config_form('', $user->id);
+        $expectedconfigbutton = 'Enable Mattermost Notifications';
+
+        $this->assertTrue(strpos($outputconfigbutton, $expectedconfigbutton) !== false);
+    }
+
+    /**
+     * Function to test update_preference when notification preference is enabled
+     */
+    public function test_update_preference_with_notification_preference_enabled() {
+        global $DB;
+        $mattermostmanager = new message_mattermost\manager();
+
+        // Create a new test user.
+        $user = $this->create_test_user_and_reset_state();
+
+        $mattermostmanager->update_preference(1, $user);
+        $outputpreferences = $DB->get_record('user_preferences', array(
+            'userid' => $user->id
+        ));
+
+        $this->assertTrue($outputpreferences->value == 1);
+    }
+
+    /**
+     * Function to test update_preference when notification preference is disabled
+     */
+    public function test_update_preference_with_notification_preference_disabled() {
+        global $DB;
+        $mattermostmanager = new message_mattermost\manager();
+
+        // Create a new test user.
+        $user = $this->create_test_user_and_reset_state();
+
+        $mattermostmanager->update_preference(false, $user);
+        $outputpreferences = $DB->get_record('user_preferences', array(
+            'userid' => $user->id
+        ));
+
+        $this->assertFalse($outputpreferences->value == 1);
```

Comment:
Should this be 0?

Reply:
As per current implementation at Moodle side this value is set to `1` in DB when `true` otherwise its set to `empty`, so here we are considering `false` case by checking for value != 1.